src/Controllers/LinkFieldController.php

```diff
@@ -274,17 +291,56 @@ private function linkFromRequest(): Link
         return $link;
     }
 
+    /**
+     * Get all Link objects based on the itemID query string argument
+     */
+    private function linksFromRequest(): DataList
+    {
+        $itemIDs = $this->itemIDsFromRequest();
+        if (empty($itemIDs)) {
+            $this->jsonError(404, _t('LinkField.INVALID_ID', 'Invalid ID'));
+        }
+        $link = Link::get()->byIDs($itemIDs);
+        if (!$link->exists()) {
+            $this->jsonError(404, _t('LinkField.INVALID_ID', 'Invalid ID'));
+        }
+        return $link;
```

Comment:
```suggestion
        $links = Link::get()->byIDs($itemIDs);
        if (!$links->exists()) {
            $this->jsonError(404, _t('LinkField.INVALID_ID', 'Invalid ID'));
        }
        return $links;
```
Though I think the use of DataList::exists() isn't quite right, since that will `Returns true if this DataList has items` - need to check the number of items returns is the same as the length of $itemIDs

Reply:
Updated to `$links`. Also, `exists()` is correct here because we do only care if there are items in the list, not how many items there are. We explicitly passed in the array of IDs from the request. There will never be _more_ than that number of links. If there are fewer, it means that the relation was outdated and had some links in it that don't exist. Only the links that exist will be displayed, and when the user clicks save the missing link IDs will disappear.